### PR TITLE
#16 mute output

### PR DESCRIPTION
**Pull request by kaklakariada**
See #16

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Changelog
```
Comment (redcatbear) [inline CHANGELOG.md]: Please keep the Change log in the GitHub releases to have a single source of truth.

File: src/main/java/org/itsallcode/io/CapturingOutputStream.java [hunk 2]
```diff
@@ -88,6 +89,13 @@ public void capture()
         this.internalStream = new ByteArrayOutputStream();
     }
 
+    @Override
+    public void captureMuted()
```
Comment (redcatbear) [inline src/main/java/org/itsallcode/io/CapturingOutputStream.java:93]: This solution is refreshingly primitive. I love it.

And here I was thinking about complicated configuration methods. :-)

File: src/test/java/org/itsallcode/io/TestCapturingOuputStream.java [hunk 3]
```diff
@@ -55,4 +70,39 @@ void testWithBufferedOutputStreamOnTopOfSystemOut() throws IOException
             System.setOut(previousOut);
         }
     }
+
+    @Test
+    void testOutputForwardedWhenCapturing() throws IOException
+    {
+        final OutputStream delegate = mock(OutputStream.class);
```
Comment (redcatbear) [inline src/test/java/org/itsallcode/io/TestCapturingOuputStream.java]: If your reduce the code duplication between this and the next method, the difference might be easier to spot.

File: src/test/java/org/itsallcode/io/TestCapturingOuputStream.java [hunk 4]
```diff
@@ -55,4 +70,39 @@ void testWithBufferedOutputStreamOnTopOfSystemOut() throws IOException
...
+        final CapturingOutputStream stream = new CapturingOutputStream(delegate);
+        stream.capture();
+        final byte[] content = "content".getBytes();
+        stream.write(content);
+        stream.close();
+        verify(delegate).write(content, 0, 7);
```
Comment (redcatbear) [inline src/test/java/org/itsallcode/io/TestCapturingOuputStream.java]: Use length of content as third parameter to express the intent better.

File: src/test/java/org/itsallcode/io/TestCapturingOuputStream.java [hunk 5]
```diff
@@ -55,4 +70,39 @@ void testWithBufferedOutputStreamOnTopOfSystemOut() throws IOException
...
+    }
+
+    @Test
+    void testOutputNotForwardedWhenCapturingInMuteMode() throws IOException
+    {
+        final OutputStream delegate = mock(OutputStream.class);
+        final CapturingOutputStream stream = new CapturingOutputStream(delegate);
+        stream.captureMuted();
+        final byte[] content = "content\n".getBytes();
```
Comment (redcatbear) [inline src/test/java/org/itsallcode/io/TestCapturingOuputStream.java]: Is there a special reason why you add the newline here in contrast to the tests above?

File: src/test/java/org/itsallcode/io/TestCapturingOuputStream.java [hunk 6]
```diff
@@ -55,4 +70,39 @@ void testWithBufferedOutputStreamOnTopOfSystemOut() throws IOException
...
+        final byte[] content = "content".getBytes();
+        stream.write(content);
+        stream.close();
+        verify(delegate).write(content, 0, 7);
+    }
+
+    @Test
+    void testOutputForwardedWhenNotCapturing() throws IOException
+    {
+        final OutputStream delegate = mock(OutputStream.class);
```
Comment (redcatbear) [inline src/test/java/org/itsallcode/io/TestCapturingOuputStream.java]: Using mocking here is a very white-box-style test for implementation in an external library. On the other hand not doing this with a mock will likely create an infinite loop of trying self-proof. Accepted.

File: .travis.yml [hunk 7]
```diff
@@ -18,10 +18,3 @@ script:
 
 after_success:
 - codecov
```
Comment (redcatbear) [inline .travis.yml:20]: Something important seems to be missing here. Please double check!

**Review by redcatbear — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 8]
```diff
@@ -64,6 +64,12 @@
             <version>${junit.platform.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (redcatbear) [inline pom.xml:67]: Please document new OSS dependency in the README for attribution.

**Review by redcatbear — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 7 above)
Comment (kaklakariada) [reply .travis.yml:20]: It's ok like this.

File: CHANGELOG.md (hunk 1 above)
Comment (kaklakariada) [reply CHANGELOG.md]: ✅

File: pom.xml (hunk 8 above)
Comment (kaklakariada) [reply pom.xml:67]: ✅

File: src/test/java/org/itsallcode/io/TestCapturingOuputStream.java (hunk 3 above)
Comment (kaklakariada) [reply src/test/java/org/itsallcode/io/TestCapturingOuputStream.java]: ✅

File: src/test/java/org/itsallcode/io/TestCapturingOuputStream.java (hunk 4 above)
Comment (kaklakariada) [reply src/test/java/org/itsallcode/io/TestCapturingOuputStream.java]: ✅

File: src/test/java/org/itsallcode/io/TestCapturingOuputStream.java (hunk 5 above)
Comment (kaklakariada) [reply src/test/java/org/itsallcode/io/TestCapturingOuputStream.java]: No. I fixed this. ✅

**Review by redcatbear — APPROVED** [review]
Approved. Thanks for the changes.

**Pull request merged**